source/server/server.cc

```diff
@@ -749,7 +748,12 @@ absl::Status InstanceBase::initializeOrThrow(Network::Address::InstanceConstShar
   }
 
   // Once we have runtime we can initialize the SSL context manager.
-  ssl_context_manager_ = createContextManager("ssl_context_manager", time_source_);
+  Ssl::ContextManagerFactory* ssl_factory =
+      Registry::FactoryRegistry<Ssl::ContextManagerFactory>::getFactory("ssl_context_manager");
+  if (!ssl_factory) {
+    throwEnvoyExceptionOrPanic("No SSL factory compiled");
```

Comment:
Isn't this a regression? Before this change, it was possible (I think?) to compile Envoy with no TLS stack at all, and it would mostly work. With this change, it makes it throw an error on startup. 

I don't know if anyone actually uses Envoy with no TLS stack or not.

Reply:
so if you want to compile Enovy with no TLS stack, you compile in the contrib stub extension instead.  It now registers as the basic ssl_context_manager

Reply:
I think I'm missing some context for why this change is being made. It's a tiny chunk of code. What's the advantage of moving it into contrib, vs leaving it where it is?

Reply:
I dislike the approach because you can build an envoy binary that cannot start; that should be a compile-time error.

However, I experimented a bit with no-ssl builds of envoy this morning. I tried `bazel build  --config=clang  -- //:envoy --define boringssl=disabled --//bazel:http3=False --//source/extensions/transport_sockets/tls:enabled=false`. What I got was a build with no tls transport socket, `envoy --version` says `065710fbe7f453d7f8ba44871330af33b0a0d978/1.30.0-dev/Modified/DEBUG/no-ssl`, but `nm` clearly shows all of the boringssl symbols compiled in. I have no idea what in the build system caused that, but it seems clear that nobody is using a no-ssl build of envoy. @phlax are you aware of anyone building Envoy without ssl support?

If it can't even be built without ssl, can we make this not-an-extension, and just have the server use the one and only ssl context manager statically?

Reply:
My other finding is that the methods that throw (create context) are only called from the tls transport socket, and from the quic transport socket (which depends at compiletime on the tls transport socket). So if you compile out the tls transport socket, the code that throws will never be called.

So I wonder if the interfaces can be refactored so that the fallback code doesn't need to implement that function at all. Really the only part that needs to be there is returning information about next cert to expire and similar, and that can already return an empty optional.

Reply:
if it's simply not working I'm fine just removing it. opened another PR up for that just in case they want to retcon at some point